### PR TITLE
Make 'type' field required in every component configuration.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Make 'type' field required in every component configuration.

## Fixes

Partially addresses #1702.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
n/a

### CODE REVIEW

File: aea/configurations/loader.py [hunk 1]
```diff
@@ -252,34 +252,14 @@ def from_configuration_type(
         configuration_type = PackageType(configuration_type)
         return ConfigLoaders.from_package_type(configuration_type)
 
-    def _validate(self, json_data: Dict) -> None:
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:255]: No longer used?

File: tests/test_configurations/test_loader.py [hunk 2]
```diff
@@ -71,20 +71,6 @@ def test_config_loader_dump_agent_config():
         config_loader.dump(configuration, open("foo"))
 
 
-@mock.patch.object(
-    aea.configurations.loader, "yaml_load", return_value=dict(type="connection")
-)
-def test_load_wrong_type(_):
```
Comment (DavidMinarsch) [inline tests/test_configurations/test_loader.py:77]: I guess now unreachable?

File: aea/configurations/loader.py (hunk 1 above)
Comment (marcofavorito) [reply aea/configurations/loader.py:255]: yeah, the only reason was to check the 'type' field. Now the check is incorporated in the JSON schema validation.

File: tests/test_configurations/test_loader.py (hunk 2 above)
Comment (marcofavorito) [reply tests/test_configurations/test_loader.py:77]: Yes

**Review by DavidMinarsch — COMMENTED** [review]
You'll have to fetch newer package versions from the registry to avoid these tests failing

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**